sessions/forecasting-concepts.qmd

```diff
@@ -24,13 +24,644 @@ library("ggplot2")
+
+::: {.callout-tip}
+## Interpreting the PIT histogram
+- Ideally PIT histograms should be uniform. 
+- If is a U shape then the model is overconfident and if it is an inverted U shape then the model is underconfident. 
+- If it is skewed then the model is biased towards the direction of the skew.
+:::
+
+::: {.callout-tip, collapse="true"}
+## Mathematical Definition (optional)
+
+### Continuous Case
+
+For a continuous random variable $ X $ with cumulative distribution function (CDF) $ F_X $, the PIT is defined as:
+
+$ Y = F_X(X) $
+
+where $ Y $ is uniformly distributed on $[0, 1]$. 
+
+#### Integer Case
+
+When dealing with integer forecasts, the standard PIT does not yield a uniform distribution even if the forecasts are perfectly calibrated. To address this, a randomized version of the PIT is used. For an integer-valued random variable $ X $ with CDF $ F_X $, the randomized PIT is defined as:
+
+$$ U = F_X(k) + v \cdot (F_X(k) - F_X(k-1)) $$
+
+where:
+- $ k $ is the observed integer value.
+- $ F_X(k) $ is the CDF evaluated at $ k $.
+- $ v $ is a random variable uniformly distributed on $[0, 1]$.
+
+This transformation ensures that $ U $ is uniformly distributed on $[0, 1]$ if the forecasted distribution $ F_X $ is correctly specified.
+:::
+
+Let's first look at the overall PIT histogram.
+
+```{r pit-histogram}
+ sc_forecasts |>
+  get_pit(by = "model") |>
+  plot_pit()
+```
+
+As before lets look at the PIT histogram by forecast horizon (to save space we will group horizons)
+
+```{r pit-histogram-horizon}
+sc_forecasts |> 
+  mutate(group_horizon = case_when(
+    horizon <= 3 ~ "1-3",
+    horizon <= 7 ~ "4-7",
+    horizon <= 14 ~ "8-14"
+  )) |>
+  get_pit(by = "group_horizon") |>
+  plot_pit() +
+  facet_wrap(~group_horizon)
+```
+
+and then for different forecast dates.
+
+```{r pit-histogram-date}
+ sc_forecasts |>
+  get_pit(by = "target_day") |>
+  plot_pit() +
+  facet_wrap(~target_day)
+```
+
+::: {.callout-tip}
+## Take 5 minutes
+What do you think of the PIT histograms?
+Do they look well calibrated?
+Do they look biased?
+:::
+
+::: {.callout-note collapse="true"}
+## Solution
+- It looks like the model is biased towards over predicting and that this bias gets worse at longer forecast horizons.
+- Looking over forecast dates it looks like much of  this bias is coming from near the outbreak peak where the model is consistently over predicting but the model is also over predicting at other times.
+:::
+
+## Scoring on the log scale
+
+We can also score on the log scale.
+This can be useful if we are interested in the relative performance of the model at different scales of the data.
+For example, if we are interested in the model's performance at capturing the exponential growth phase of the epidemic.
```

Comment:
```suggestion
This can be useful if we are interested in the relative performance of the model at different scales of the data, for example if we are interested in the model's performance at capturing the exponential growth phase of the epidemic.
```